### DISCUSSION TITLE
Fast liquidity v2

### DISCUSSION THREAD

**Discussion by prestwich** [celo-org/optics-monorepo#596] [Ideas]
Plans:
1. change the `_prefill` to not allow transfer actions
2. add a new action type `TransferV2` or something
   - allow user to set their fee (between 0 and 10,000)
3. add handling for user-set fees in prefill
4. deploy as an upgrade to the bridge router

TODO: 
- human action to ensure that bridge router upgrades happen relatively synchronously?
- e.g. router A dispatches a TransferV2 before Router B is equipped to handle them

**Comment by pranaymohan** [comment]
Maybe more like a v3 idea, but was wondering if it would be feasible to do pooled LP style fast liquidity? Ie. many LPs contribute assets to an AMM style pool, and token bridge transfers get filled automatically if the pool contains the representation. LPs split fees pro rata based on tokens contributed. 

Pros:
- easier for small LPs -- no need to have hot key and manually sign a tx to prefill FL
- likely good for short tail tokens where there's enough depth and turnover
- visibility of pool depth is useful:
  - a priori if a tx will get filled -- user also can inspect pools and know
  - can dynamically drive liquidity to representations if capital is clearing or being removed 
- representation pools can set custom fees

Cons:
- capital inefficient
- long tail tokens may not have enough volume to justify passive contribution
- extra gas cost paid by processor?
- could be subject to DOS?
